### PR TITLE
[JEP-305] Test incrementals

### PR DESCRIPTION
**Pull request by batmat**
Similar test to https://github.com/jenkinsci/git-plugin/pull/582

cc @jglick FYI

### CODE REVIEW

**Comment by batmat** [comment]
expected to see `buildtriggerbadge` deployed to https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/

**Comment by batmat** [comment]
```
[ldtriggerbadge-plugin_PR-27-TGZTMUAPG3HZWBSW66YZHPGPLLWT37QZ6EWEMA5T67SSFUAHD7EQ] Running shell script

+ curl --fail -d {"build_url":"https://ci.jenkins.io/job/Plugins/job/buildtriggerbadge-plugin/job/PR-27/1/"} https://jenkins-infra-functions.azurewebsites.net/api/incrementals-publisher?code=****

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    91    0     0  100    91      0     65  0:00:01  0:00:01 --:--:--    65
100    91    0     0  100    91      0     38  0:00:02  0:00:02 --:--:--    38
100    91    0     0  100    91      0     26  0:00:03  0:00:03 --:--:--    26
100    91    0     0  100    91      0     20  0:00:04  0:00:04 --:--:--    20
100    91    0     0  100    91      0     16  0:00:05  0:00:05 --:--:--    18
100    91    0     0  100    91      0     14  0:00:06  0:00:06 --:--:--     0
100    91    0     0  100    91      0     12  0:00:07  0:00:07 --:--:--     0
100    91    0     0  100    91      0     10  0:00:09  0:00:08  0:00:01     0
100    91    0     0  100    91      0      9  0:00:10  0:00:09  0:00:01     0
100    91    0     0  100    91      0      8  0:00:11  0:00:10  0:00:01     0
100    91    0     0  100    91      0      7  0:00:13  0:00:11  0:00:02     0
100    91    0     0  100    91      0      7  0:00:13  0:00:12  0:00:01     0
100    91    0     0  100    91      0      6  0:00:15  0:00:13  0:00:02     0
100    91    0     0  100    91      0      6  0:00:15  0:00:14  0:00:01     0
100    91    0     0  100    91      0      5  0:00:18  0:00:15  0:00:03     0
100    91    0     0  100    91      0      5  0:00:18  0:00:16  0:00:02     0
100    91    0     0  100    91      0      5  0:00:18  0:00:17  0:00:01     0
100    91    0     0  100    91      0      4  0:00:22  0:00:18  0:00:04     0
100    91    0     0  100    91      0      4  0:00:22  0:00:19  0:00:03     0
100    91    0     0  100    91      0      4  0:00:22  0:00:20  0:00:02     0
100    91    0     0  100    91      0      4  0:00:22  0:00:21  0:00:01     0
100    91    0     0  100    91      0      4  0:00:22  0:00:22 --:--:--     0
100    91    0     0  100    91      0      3  0:00:30  0:00:23  0:00:07     0
100    91    0     0  100    91      0      3  0:00:30  0:00:24  0:00:06     0
100    91    0     0  100    91      0      3  0:00:30  0:00:25  0:00:05     0
100    91    0     0  100    91      0      3  0:00:30  0:00:26  0:00:04     0
100    91    0     0  100    91      0      3  0:00:30  0:00:27  0:00:03     0
100    91    0     0  100    91      0      3  0:00:30  0:00:28  0:00:02     0
100    91    0     0  100    91      0      3  0:00:30  0:00:29  0:00:01     0
100    91    0     0  100    91      0      2  0:00:45  0:00:30  0:00:15     0
100    91    0     0  100    91      0      2  0:00:45  0:00:31  0:00:14     0
100    91    0     0  100    91      0      2  0:00:45  0:00:32  0:00:13     0
100    91    0     0  100    91      0      2  0:00:45  0:00:33  0:00:12     0
100    91    0     0  100    91      0      2  0:00:45  0:00:34  0:00:11     0
100    91    0     0  100    91      0      2  0:00:45  0:00:35  0:00:10     0
100    91    0     0  100    91      0      2  0:00:45  0:00:36  0:00:09     0
100    91    0     0  100    91      0      2  0:00:45  0:00:37  0:00:08     0
100    91    0     0  100    91      0      2  0:00:45  0:00:38  0:00:07     0
100    91    0     0  100    91      0      2  0:00:45  0:00:39  0:00:06     0
100    91    0     0  100    91      0      2  0:00:45  0:00:40  0:00:05     0
100    91    0     0  100    91      0      2  0:00:45  0:00:41  0:00:04     0
100    91    0     0  100    91      0      2  0:00:45  0:00:42  0:00:03     0
100    91    0     0  100    91      0      2  0:00:45  0:00:43  0:00:02     0
100    91    0     0  100    91      0      2  0:00:45  0:00:44  0:00:01     0
100    91    0     0  100    91      0      2  0:00:45  0:00:45 --:--:--     0
100    91    0     0  100    91      0      1  0:01:31  0:00:46  0:00:45     0
100    91    0     0  100    91      0      1  0:01:31  0:00:47  0:00:44     0
100    91    0     0  100    91      0      1  0:01:31  0:00:48  0:00:43     0
100    91    0     0  100    91      0      1  0:01:31  0:00:49  0:00:42     0
100    91    0     0  100    91      0      1  0:01:31  0:00:50  0:00:41     0
100    91    0     0  100    91      0      1  0:01:31  0:00:51  0:00:40     0
100    91    0     0  100    91      0      1  0:01:31  0:00:52  0:00:39     0
100    91    0     0  100    91      0      1  0:01:31  0:00:53  0:00:38     0
100    91    0     0  100    91      0      1  0:01:31  0:00:54  0:00:37     0
100    91    0     0  100    91      0      1  0:01:31  0:00:55  0:00:36     0
100    91    0     0  100    91      0      1  0:01:31  0:00:56  0:00:35     0
100    91    0     0  100    91      0      1  0:01:31  0:00:57  0:00:34     0
100    91    0     0  100    91      0      1  0:01:31  0:00:58  0:00:33     0
100    91    0     0  100    91      0      1  0:01:31  0:00:59  0:00:32     0
100    91    0     0  100    91      0      1  0:01:31  0:01:00  0:00:31     0
100    91    0     0  100    91      0      1  0:01:31  0:01:01  0:00:30     0
100    91    0     0  100    91      0      1  0:01:31  0:01:02  0:00:29     0

curl: (22) The requested URL returned error: 500 Internal Server Error

script returned exit code 22
```

ahem.

**Comment by jglick** [comment]
Forgot to mention that pending [INFRA-1598](https://issues.jenkins-ci.org/browse/INFRA-1598), you need to patch `repository-permissions-updater` for your plugin.

File: pom.xml [hunk 1]
```diff
@@ -3,11 +3,11 @@
 	<parent>
 		<groupId>org.jenkins-ci.plugins</groupId>
 		<artifactId>plugin</artifactId>
-		<version>3.8</version>
+		<version>3.9-20180419.172824-3</version>
```
Comment (jglick) [inline pom.xml]: 3.9 released

File: pom.xml [hunk 2]
```diff
@@ -31,13 +31,17 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<jenkins.version>1.625.3</jenkins.version>
 		<java.level>7</java.level>
+
+		<revision>1.23</revision>
```
Comment (jglick) [inline pom.xml]: `2.10`!

**Comment by batmat** [comment]
image

image

\o/ great @jglick !

**Comment by batmat** [comment]
Let's merge. I'll try to release a normal version to double-check how this behaves when one does mix incrementals and _historically-normal :-)_ releases.

**Pull request merged**